src/edu/buffalo/cse/jive/finiteStateMachine/monitor/Monitor.java

```diff
@@ -114,9 +118,12 @@ public void resetStates() {
 	 */
 	private boolean validate(State root, List<Expression> expressions) {
 		boolean valid = true;
+		Context thisContext = new Context(root, null, states);
+		if(expressions.get(0) instanceof EExpression)thisContext = new Context(root, null, states,true);

```

Comment:
This line makes sure if E expression exists on the top level only then we will try to find the shortest path. Otherwise, it will use the old method for evaluating E Expressions.